zulipterminal/core.py

```diff
@@ -392,8 +392,10 @@ def _narrow_to(self, anchor: Optional[int], **narrow: Any) -> None:
 
         msg_id_list = self.model.get_message_ids_in_current_narrow()
 
-        # if no messages are found get more messages
-        if len(msg_id_list) == 0:
+        # If no messages are found in the current narrow
+        # OR, given anchor is not present in msg_id_list
+        # then, get more messages.
+        if len(msg_id_list) == 0 or (anchor is not None and anchor not in msg_id_list):
```

Comment:
Worth linking to the discussion in the commit text, perhaps.

Reply:
To clarify, do you mean adding a link (or commit-hash?) to this commit?

Reply:
I've added the commit hash in the comments. I am not sure if I can add a link the commit?